torch/csrc/cudnn/Conv.cpp

```diff
@@ -220,68 +303,119 @@ struct algorithm_search<cudnnConvolutionBwdDataAlgo_t> {
-         CUDNN_CONVOLUTION_BWD_FILTER_ALGO_FFT_TILING,
+        CUDNN_CONVOLUTION_BWD_FILTER_ALGO_FFT_TILING,
 #endif
     };
-    size_t max_ws_size = getMaxWorkspaceSize<cudnnConvolutionBwdFilterAlgo_t>(handle,conv,algo,sizeof(algo)/sizeof(algo[0]),state);
+    size_t max_ws_size = getMaxWorkspaceSize<cudnnConvolutionBwdFilterAlgo_t>(
+        handle, conv, algo, sizeof(algo)/sizeof(algo[0]), state);
     Workspace ws(state, max_ws_size);
 
-    CHECK(cudnnFindConvolutionBackwardFilterAlgorithmEx(handle, conv.idesc.desc, in,
-        conv.odesc.desc, out, conv.cdesc.desc, conv.wdesc.desc, wght, 1, &algoCount,
-        &perfResults, ws.data, ws.size));
+    CHECK(cudnnFindConvolutionBackwardFilterAlgorithmEx(
+        handle,
+        conv.idesc.desc,
+        in,
+        conv.odesc.desc,
+        out,
+        conv.cdesc.desc,
+        conv.wdesc.desc,
+        wght,
+        1,
+        &algoCount,
+        &perfResults,
+        ws.data,
+        ws.size));
 
     return perfResults;
   }
 
-  static void getAlgorithm(cudnnHandle_t handle, const Convolution& conv, cudnnConvolutionBwdFilterAlgo_t* algo) {
-    CHECK(cudnnGetConvolutionBackwardFilterAlgorithm(handle, conv.idesc.desc,
-        conv.odesc.desc, conv.cdesc.desc, conv.wdesc.desc,
-        CUDNN_CONVOLUTION_BWD_FILTER_PREFER_FASTEST, 0, algo));
-  }
-
-  static void getWorkspaceSize(cudnnHandle_t handle, const Convolution& conv, cudnnConvolutionBwdFilterAlgo_t algo, size_t* workspaceSize) {
-    CHECK(cudnnGetConvolutionBackwardFilterWorkspaceSize(handle, conv.idesc.desc,
-        conv.odesc.desc, conv.cdesc.desc, conv.wdesc.desc, algo, workspaceSize));
+  static void getAlgorithm(
+      cudnnHandle_t handle, const Convolution& conv,
+      cudnnConvolutionBwdFilterAlgo_t* algo)
+  {
+    CHECK(cudnnGetConvolutionBackwardFilterAlgorithm(
+        handle,
+        conv.idesc.desc,
+        conv.odesc.desc,
+        conv.cdesc.desc,
+        conv.wdesc.desc,
+        CUDNN_CONVOLUTION_BWD_FILTER_PREFER_FASTEST,
+        0,
+        algo)
+    );
+  }
+
+  static void getWorkspaceSize(
+      cudnnHandle_t handle, const Convolution& conv,
+      cudnnConvolutionBwdFilterAlgo_t algo, size_t* workspaceSize)
+  {
+    CHECK(cudnnGetConvolutionBackwardFilterWorkspaceSize(
+        handle,
+        conv.idesc.desc,
+        conv.odesc.desc,
+        conv.cdesc.desc,
+        conv.wdesc.desc,
+        algo,
+        workspaceSize));
   }
 };
 
 template<typename algo_t>
 void findAlgorithm(
     THCState* state, cudnnHandle_t handle, const Convolution& conv,
-    bool benchmark, void* in, void* out, void* wght, algo_t* algo)
+    bool benchmark, bool deterministic, void* in, void* out, void* wght,
+    algo_t* algo)
 {
+  // declare an instance of the algo class {fwd, bwd_filter, bwd_data} that we
+  // want to search on
   using search = algorithm_search<algo_t>;
   auto& cache = search::cache();
 
   if (cache.find(conv.params, algo)) {
     return;
   }
 
+  if (deterministic) {
```

Comment:
this is the relevant change in this file for determinism however I found the code slightly hard to read so I made some style changes. If people don't like them I can revert as well but I find this code much more readable now